rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/CatalogResource.java

```diff
@@ -261,15 +245,37 @@ public Response createFromArchive(byte[] zipInput) {
                     throw new IllegalArgumentException("Error installing catalog items", ex);
                 }
             }
-            
-            return Response.status(Status.CREATED).build();
+
+            // TODO improve on this - If the FEATURE_LOAD_BUNDLE_CATALOG_BOM is enabled, the REST API won't return the
+            // added items which breaks the contract on the API endpoint.
+            return buildCreateResponse(catalogItems);
         } catch (RuntimeException ex) {
             throw WebResourceUtils.badRequest(ex);
         } finally {
             if (f!=null) f.delete();
             if (f2!=null) f2.delete();
         }
     }
+
+    private Response buildCreateResponse(Iterable<? extends CatalogItem<?, ?>> catalogItems) {
+        log.info("REST created catalog items: "+catalogItems);
+
+        Map<String,Object> result = MutableMap.of();
+
+        for (CatalogItem<?,?> catalogItem: catalogItems) {
+            try {
+                result.put(catalogItem.getId(), CatalogTransformer.catalogItemSummary(brooklyn(), catalogItem, ui.getBaseUriBuilder()));
+            } catch (Throwable t) {
+                log.warn("Error loading catalog item '"+catalogItem+"' (rethrowing): "+t);
+                // unfortunately items are already added to the catalog and hard to remove,
+                // but at least let the user know;
+                // happens eg if a class refers to a missing class, like
+                // loading nosql items including mongo without the mongo bson class on the classpath
+                throw Exceptions.propagateAnnotated("At least one unusable item was added ("+catalogItem.getId()+")", t);
```

Comment:
@geomacy I move this code from the `createFromYaml` function, agree that it could be refactored but It is not in the scope of this PR. I would leave it for future improvement